utils/http.go

```diff
@@ -112,6 +176,21 @@ func buildAccessRecords(repo string, actions ...string) []auth.Access {
 	return requiredAccess
 }
 
+// buildCatalogRecord returns the only valid format for the catalog
+// resource. Only admins can get this access level from the token
+// server.
+func buildCatalogRecord(actions ...string) []auth.Access {
+	requiredAccess := []auth.Access{{
+		Resource: auth.Resource{
+			Type: "registry",
```

Comment:
Why use a different `Type` here with what we have in `buildAccessRecords()`, and does the `Type` matter to the whole auth process?